tools/topology/topology2/development/cavs-nocodec-dp-test.conf

```diff
@@ -0,0 +1,1702 @@
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           16
+									out_valid_bit_depth     16
+								}
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           32
+									out_valid_bit_depth     24
+								}
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           32
+									out_valid_bit_depth     32
+								}
+							]
+						}
+					}
+				]
+				Object.Pipeline.host-gateway-playback [
+					{
+						index	3
+						Object.Widget.host-copier.1 {
+							stream_name 'SSP1 Playback'
+							pcm_id 1
+							num_input_audio_formats 3
+							Object.Base.input_audio_format [
+								{
+									in_rate         $SSP1_RATE
+									in_bit_depth            16
+									in_valid_bit_depth      16
+								}
+								{
+									in_rate         $SSP1_RATE
+									in_bit_depth            32
+									in_valid_bit_depth      24
+								}
+								{
+									in_rate         $SSP1_RATE
+									in_bit_depth            32
+									in_valid_bit_depth      32
+								}
+							]
+							num_output_audio_formats 3
+							Object.Base.output_audio_format [
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           16
+									out_valid_bit_depth     16
+								}
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           32
+									out_valid_bit_depth     24
+								}
+								{
+									out_rate        $SSP1_RATE
+									out_bit_depth           32
+									out_valid_bit_depth     32
+								}
+							]
+						}
+					}
+				]
+			}
+		}
+
+		Object.PCM.pcm [
+			{
+				name	"$SSP1_PCM_NAME"
+				id 1
+				direction	"duplex"
+				Object.Base.fe_dai.1 {
+					name	"$SSP1_PCM_NAME"
+				}
+
+				Object.PCM.pcm_caps.1 {
+					direction	"playback"
+					name "SSP1 Playback"
+					formats 'S16_LE,S24_LE,S32_LE'
+					IncludeByKey.SSP1_RATE {
+						"48000" {
+							rates '48000'
+						}
+						"96000" {
+							rates '96000'
+						}
+						"192000" {
+							rates '192000'
+						}
+					}
+				}
+
+				Object.PCM.pcm_caps.2 {
+					direction	"capture"
+					name "SSP1 Capture"
+					formats 'S16_LE,S24_LE,S32_LE'
+					IncludeByKey.SSP1_RATE {
+						"48000" {
+							rates '48000'
+						}
+						"96000" {
+							rates '96000'
+						}
+						"192000" {
+							rates '192000'
+						}
+					}
+				}
+			}
+		]
+
+		IncludeByKey.PASSTHROUGH {
+		"false" {
+				Object.Base.route [
+					{
+						source	"mixin.3.1"
+						sink	"mixout.4.1"
+					}
+					{
+						source	"gain.4.1"
+						sink	"dai-copier.SSP.NoCodec-1.playback"
+					}
+					{
+						source 'host-copier.1.playback'
+						sink 'gain.3.1'
+					}
+				]
+			}
+		"true"	{
+			Object.Base.route [
+				{
+					source	"host-copier.1.playback"
+					sink	"dai-copier.SSP.NoCodec-1.playback"
+				}
+			]
+			}
+		}
+		Object.Base.route [
+			{
+				source	"dai-copier.SSP.NoCodec-1.capture"
+				sink	"host-copier.1.capture"
+			}
+		]
+	}
+}
```

Comment:
I wonder if this could reuse nocodec more and just override the schedule. nocodec is one of the more complex so cloning it completely is not ideal for maintainance. For BT we have a scaled down nocodec that is easier to maintain.

DId @jsarha you consider this?

Reply:
I considered, but I could not figure out a way to do it without adding even more variables and even more IncludeByKey -sections to cavs-nocodec.conf. I feelt nocodec topology is already horrible enough to read. But yes. For this particular purpose it could actually make sense even if I do not really like the way we currently use the IncludeByKey -sections at the topology top level.

But eventually we need a major restructure to the whole topology hierarchy.  More top-level files and less variation in each of them. The first in my kill list is PASSTHROUGH variable and the numerous IncludeByKey-sections sensitive to it.

But yes, I'll add one more variable and couple of more IncludeByKey-sections.